File: bot.js
hereconst puppeteer = require('puppeteer-extra');
const StealthPlugin = require('puppeteer-extra-plugin-stealth');
puppeteer.use(StealthPlugin());

async function start() {
    const browser = await puppeteer.launch({
        headless: "new",
        args: ['--no-sandbox', '--disable-setuid-sandbox']
    });
    const page = await browser.newPage();
    
    try {
        await page.goto(process.env.TARGET_URL, { waitUntil: 'domcontentloaded', timeout: 60000 });
        console.log("Page 1: Navigating...");

        // 1. Verify/Continue Button
        const vBtn = "//a[contains(text(), 'Verify')] | //button[contains(text(), 'Verify')]";
        await page.waitForXPath(vBtn, { visible: true, timeout: 20000 }).catch(() => console.log("Skip step 1"));
        const [btn1] = await page.$x(vBtn);
        if (btn1) await btn1.click();
        
        console.log("Waiting for timer (15s)...");
        await new Promise(r => setTimeout(r, 15000));

        // 2. Universal "Get Link" Finder (ID pe depend nahi karega)
        console.log("Searching for 'Get Link' button...");
        await page.waitForFunction(() => {
            const elements = Array.from(document.querySelectorAll('a, button'));
            const target = elements.find(el => el.innerText.toLowerCase().includes('get link') || el.innerText.toLowerCase().includes('open link'));
            if (target) {
                target.click();
                return true;
            }
            return false;
        }, { timeout: 20000 });

        console.log("Success! Final button clicked.");
        await new Promise(r => setTimeout(r, 5000));
        console.log("Final URL: " + page.url());

    } catch (e) {
        console.error("Critical Error: " + e.message);
    } finally {
        await browser.close();
    }
}
start();

File: bot.py
from playwright.sync_api import sync_playwright
import time
import sys

sys.stdout.reconfigure(line_buffering=True)

def run_bot():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(user_agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36")
        
        # AD-BLOCKER: Ads aur Trackers ko block karne ke liye
        ad_list = ["*doubleclick.net*", "*googleadservices.com*", "*adservice.google.com*", "*googlesyndication.com*", "*.ads.*"]
        page = context.new_page()
        
        def block_ads(route):
            if any(ad in route.request.url for ad in ad_list):
                route.abort()
            else:
                route.continue_()
        
        page.route("**/*", block_ads)
        
        target_url = "https://indianshortner.in/5Nrtc"
        print(f"[LOG] Navigating with Ad-Blocker: {target_url}")
        page.goto(target_url, wait_until="networkidle")

        def handle_step(text):
            print(f"[LOG] Finding: {text}")
            try:
                # Button dhundhna
                selector = f"//button[contains(text(), '{text}')] | //a[contains(text(), '{text}')]"
                page.wait_for_selector(selector, timeout=15000)
                page.locator(selector).first.click()
                print(f"[SUCCESS] Clicked {text}")
                page.wait_for_load_state("networkidle")
                time.sleep(3)
            except Exception as e:
                print(f"[WARNING] Step skipped: {text}")

        # Cycle Execution
        for i in range(2):
            handle_step("I am not a robot")
            handle_step("Click 2x")
            handle_step("Download Link")

        print("[LOG] Waiting for timer...")
        time.sleep(15)
        
        # Final Action
        if page.locator("#btn-main").is_visible():
            page.locator("#btn-main").click()
            print(f"[SUCCESS] Final URL: {page.url}")
        
        browser.close()

if __name__ == "__main__":
    run_bot()
